File: ventureflow-backend/src/main/java/br/com/venture/ventureflow/salesorder/model/entity/SalesOrder.java
package br.com.venture.ventureflow.salesorder.model.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "sales_orders")
public class SalesOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 20)
    private String code;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(length = 500)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private SalesOrderStatus status;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private boolean active;

    public SalesOrder() {
    }

    public SalesOrder(
            String code,
            String name,
            String description,
            SalesOrderStatus status,
            LocalDateTime createdAt
    ) {
        this.code = code;
        this.name = name;
        this.description = description;
        this.status = status;
        this.createdAt = createdAt;
        this.active = true;
    }

    public Long getId() {
        return id;
    }

    public String getCode() {
        return code;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public SalesOrderStatus getStatus() {
        return status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public boolean isActive() {
        return active;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setStatus(SalesOrderStatus status) {
        this.status = status;
    }

    public void setActive(boolean active) {
        this.active = active;
    }
}

File: venture-erp-frontend/src/features/sales-orders/types/salesOrder.ts
export type SalesOrderStatus =
  | "CREATED"
  | "IN_PROGRESS"
  | "COMPLETED"
  | "CANCELLED";

export interface CreateSalesOrderRequest {
  name: string;
  description: string;
  status: SalesOrderStatus;
  resellerId: number;
}

export interface SalesOrder {
  id: number;
  code: string;
  name: string;
  description: string;
  status: SalesOrderStatus;
  active: boolean;
  createdAt: string;
  resellerId: number;
  resellerName: string;
  assemblySupervisorId: number | null;
  assemblySupervisorName: string | null;
}

export interface SalesOrderFormData {
  name: string;
  description: string;
}

export interface UpdateSalesOrderRequest {
  name: string;
  description: string;
  status: SalesOrderStatus;
  resellerId: number;
}
